internal/provider/vapp/isolated_network_resource.go

```diff
@@ -0,0 +1,511 @@
+	guestVLAN := plan.GuestVLANAllowed.ValueBool()
+	vappNetworkName := plan.Name.ValueString()
+
+	var staticIPPools []*staticIPPoolModel
+	resp.Diagnostics.Append(plan.StaticIPPool.ElementsAs(ctx, &staticIPPools, true)...)
+
+	staticIPRanges := make([]*govcdtypes.IPRange, 0)
+	for _, staticIPPool := range staticIPPools {
+		staticIPRanges = append(staticIPRanges, &govcdtypes.IPRange{
+			StartAddress: staticIPPool.StartAddress.ValueString(),
+			EndAddress:   staticIPPool.EndAddress.ValueString(),
+		})
+	}
+
+	vappNetworkSettings := &govcd.VappNetworkSettings{
+		Name:               vappNetworkName,
+		Description:        plan.Description.ValueString(),
+		Gateway:            plan.Gateway.ValueString(),
+		NetMask:            plan.Netmask.ValueString(),
+		DNS1:               plan.DNS1.ValueString(),
+		DNS2:               plan.DNS2.ValueString(),
+		DNSSuffix:          plan.DNSSuffix.ValueString(),
+		StaticIPRanges:     staticIPRanges,
+		RetainIpMacEnabled: &retainIPMac,
+		GuestVLANAllowed:   &guestVLAN,
+	}
+
+	// Create network
+	vAppNetworkConfig, err := isolatedNetworkRef.VApp.CreateVappNetwork(vappNetworkSettings, nil)
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating vApp network", err.Error())
+		return
+	}
+
+	vAppNetwork := govcdtypes.VAppNetworkConfiguration{}
+	for _, networkConfig := range vAppNetworkConfig.NetworkConfig {
+		if networkConfig.NetworkName == vappNetworkName {
+			vAppNetwork = networkConfig
+		}
+	}
+
+	if vAppNetwork == (govcdtypes.VAppNetworkConfiguration{}) {
+		resp.Diagnostics.AddError("didn't find vApp network: %s", vappNetworkName)
+		return
+	}
+
+	// Get UUID.
+	networkID, err := govcd.GetUuidFromHref(vAppNetwork.Link.HREF, false)
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating vApp network uuid", err.Error())
+		return
+	}
+
+	id := common.NormalizeID("urn:vcloud:network:", networkID)
+
+	plan.ID = types.StringValue(id)
+
+	// Set state to fully populated data
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Read refreshes the Terraform state with the latest data.
+func (r *isolatedNetworkResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var (
+		state *isolatedNetworkResourceModel
+		diag  diag.Diagnostics
+	)
+
+	// Get current state
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	isolatedNetworkRef, errInit := state.initNetworkQuery(ctx, r.client, false)
+	if errInit != nil {
+		if errInit.Summary == ErrVAppNotFound {
+			resp.State.RemoveResource(ctx)
+		}
+		resp.Diagnostics.AddError(errInit.Summary, errInit.Detail)
+		return
+	}
+
+	if isolatedNetworkRef.VAppLocked {
+		defer isolatedNetworkRef.VAppUnlockF()
+	}
+
+	vAppNetworkConfig, err := isolatedNetworkRef.VApp.GetNetworkConfig()
+	if err != nil {
+		resp.Diagnostics.AddError("error getting vApp networks", err.Error())
+		return
+	}
```

Comment:
For preventing `nil` pointer check is `isolatedNetworkRef.VApp` is nil

Reply:
This can't return a nill without error, this is catched in the initNetworkQuery function call before.